sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala

```diff
@@ -450,8 +507,27 @@ object ScalaJSBundlerPlugin extends AutoPlugin {
           streams.value.log
         ),
 
-      webpackEmitSourceMaps in stageTask := (emitSourceMaps in stageTask).value
+      webpackEmitSourceMaps in stageTask := (emitSourceMaps in stageTask).value,
+
+      webpackMonitoredFiles in stageTask := {
+        val generatedWebpackConfigFile = (scalaJSBundlerWebpackConfig in stageTask).value
+        val customWebpackConfigFile = (webpackConfigFile in stageTask).value
+        val packageJsonFile = scalaJSBundlerPackageJson.value
+        val entries = (webpackEntries in stageTask).value
 
+        val filter = (webpackMonitoredIncludeFilter in stageTask).value
+        val dirs = (webpackMonitoredDirectories in stageTask).value
+
+        val additionalFiles = dirs.flatMap(
+          dir => (dir ** filter).get
+        )
+
+        packageJsonFile +:
+          generatedWebpackConfigFile +:
+          customWebpackConfigFile ++:
+          entries.map(_._2) ++:
+          additionalFiles
```

Comment:
I think you should also add `stageTask.value.data`, which contains the .js code produced by Scala.js (the entries contain only the launcher files, which depend on them). I don’t know how we could automate the resolution of the dependencies among .js files…